lib/ansible/modules/packaging/os/flatpak_remote.py

```diff
@@ -0,0 +1,245 @@
+    default: flatpak
+  state:
+    description:
+      - Set to C(present) will install the flatpak remote.
+      - Set to C(absent) will remove the flatpak remote.
+    choices: [ absent, present ]
+    default: present
+'''
+
+EXAMPLES = r'''
+- name: Add the Gnome flatpak remote to the system installation under the name 'gnome'.
+  flatpak_remote:
+    name: gnome
+    state: present
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    method: system
+
+- name: Remove the Gnome flatpak remote  from the user installation.
+  flatpak_remote:
+    name: gnome
+    state: absent
+
+- name: Add the flathub flatpak repository remote to the user installation.
+  flatpak_remote:
+    name: flathub
+    state: present
+    remote:  https://dl.flathub.org/repo/flathub.flatpakrepo
+    method: user
+
+- name: Remove the flathub flatpak repository remote from the system installtion.
+  flatpak_remote:
+    name: flathub
+    state: absent
+    method: system
+'''
+
+RETURN = r'''
+reason:
+  description: On failure, the output for the failure
+  returned: failed
+  type: string
+  sample: error while installing...
+name:
+  description: Remote of flatpak given for the operation
+  returned: always
+  type: string
+  sample: https://sdk.gnome.org/gnome-apps.flatpakrepo
+'''
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def add_remote(module, binary, name, remote, method):
+    """
+    Add a new remote.
+
+    returns:
+        result, type: int
+            The result status of the operation
+            Possible values:
+            0 - operation successful
+            1 - operation failed
+        output, type: str
+            Output of the operation
+            Especially helpful to know that's wrong when the operation failed
+    """
+    if module.check_mode:
+        # Check if any changes would be made but don't actually make
+        # those changes
+        module.exit_json(changed=True)
+    command = "{0} remote-add --{1} {2} {3}".format(
+        binary, method, name, remote)
+
+    return_code, output = _flatpak_command(command)
+    if return_code != 0:
+        return 1, output
+
+    return 0, output
+
+
+def remove_remote(module, binary, name, method):
+    """
+    Remove an existing remote.
+
+    returns:
+        result, type: int
+            The result status of the operation
+            Possible values:
+            0 - operation successful
+            1 - operation failed
+        output, type: str
+            Output of the operation
+            Especially helpful to know that's wrong when the operation failed
+    """
+    if module.check_mode:
+        # Check if any changes would be made but don't actually make
+        # those changes
+        module.exit_json(changed=True)
+
+    command = "{0} remote-delete --{1} --force {2} ".format(
+        binary, method, name)
+    return_code, output = _flatpak_command(command)
+    if return_code != 0:
```

Comment:
Why don't we return the return_code back to the user ?
I think we should also return the command output back to the user, especially if the command failed.